Specify `light()` only runs for pixels affected by light (#12174)
godotengine/godot-docs: tutorials/shaders/shader_reference/spatial_shader.rst

written to in the ``fragment()`` function to calculate the lighting for you (subject to the render mode).

The ``light()`` function is called for every light in every pixel. It is called within a loop for each light type.

> Keep in mind the ``light()`` function only runs for pixels that are affected by at least one light source, and is
> completely skipped for pixels that are not hit by any light and shadow. You can use this to create different behavior for
> lit and unlit pixels in your shader.

Author: ```suggestion
The ``light()`` function is only guaranteed to run for pixels that are affected by at least one light source, and
*might* be completely skipped by the renderer for pixels that are not hit by any light or shadow at all.
```

Not sure if this addition (even with changes to make it more accurate) is worthwhile of being added to the docs anymore though, since it's a little vague and not directly applicable.
